src/hotspot/share/cds/aotLogging.hpp

```diff
@@ -0,0 +1,166 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_CDS_AOTLOGGING_HPP
+#define SHARE_CDS_AOTLOGGING_HPP
+
+#include "cds/cds_globals.hpp"
+#include "cds/cdsConfig.hpp"
+#include "logging/log.hpp"
+
+// UL Logging for AOT
+// ==================
+//
+// The old "CDS" feature is rebranded as "AOT" in JEP 483. Therefore, UL logging
+// related to the AOT features should be using the [aot] tag.
+//
+// However, some old scripts may be using -Xlog:cds for diagnostic purposes. To
+// provide a fair amount of backwards compatibility for such scripts, some AOT
+// logs that are likely to be used by such scripts are printed using the macros
+// in this header file.
+//
+// NOTE: MOST of the AOT logs will be using the usual macros such as log_info(aot)(...).
+// The information below does NOT apply to such logs.
+//
+// CDS Compatibility Logs & Compatibility Macros
+// =============================================
+//
+// A subset of the original CDS logs (the "CDS Compatibility Logs") have been
+// selected in JDK 25. These logs are guarded using the aot_log_xxx compatibility
+// macros. Before JDK 25, such code looked like this:
+//
+//      log_info(cds)("trying to map %s%s", info, _full_path);
+//      log_warning(cds)("Unable to read the file header.");
+//
+// New code since JDK 25:
+//
+//      aot_log_info(aot)("trying to map %s%s", info, _full_path);
+//      aot_log_warning(aot)("Unable to read the file header.");
+//
+// The messages printed with the log_aot_xxx() macros work as if they are
+// using the [cds] tag when running with the "classic" CDS flags such as
+// -XX:SharedArchiveFile:
+//
+//      $ java -Xlog:cds -XX:SharedArchiveFile=bad.jsa ...
+//      [0.020s][info][cds] trying to map bad.jsa
+//      [0.020s][warning][cds] Unable to read the file header
+//
+// However, when running new AOT flags such as-XX:AOTCache, these messages are
+// under the [aot] tag:
+//
+//      $ java -Xlog:aot -XX:AOTCache=bad.aot ...
+//      [0.020s][info][aot] trying to map bad.aot
+//      [0.020s][warning][aot] Unable to read the file header
+//
+// Rules on selection and printing
+//
+// [1] When using AOT cache
+//     - These logs can be selected ONLY with -Xlog:aot. They are always printed with [aot] decoration
+//
+// [2] When using CDS archives
+//     - These logs can be selected ONLY with -Xlog:cds. They are always printed with [cds] decoration
```

Comment:
OK, so if I use `-Xlog:all`, what will they be selected as?

Reply:
`-Xlog:all` will enable all `LogTagSets`. Each `aot_log_xxx()` statement will print exactly one log. When using AOT caches, the log will be printed with `[aot]` decorations. When using CDS archives, the log will be printed with `[cds]` decorations.